pkg/xds/context/destination_index.go

```diff
@@ -0,0 +1,150 @@
+package context
+
+import (
+	"fmt"
+
+	common_api "github.com/kumahq/kuma/api/common/v1alpha1"
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/kri"
+	"github.com/kumahq/kuma/pkg/core/resources/apis/core"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	meshexternalservice_api "github.com/kumahq/kuma/pkg/core/resources/apis/meshexternalservice/api/v1alpha1"
+	meshmzservice_api "github.com/kumahq/kuma/pkg/core/resources/apis/meshmultizoneservice/api/v1alpha1"
+	meshservice_api "github.com/kumahq/kuma/pkg/core/resources/apis/meshservice/api/v1alpha1"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/plugins/policies/core/rules/resolve"
+)
+
+type Destination interface {
+	GetPorts() []core.Port
+	FindPortByName(name string) (core.Port, bool)
+}
+
+type DestinationIndex struct {
```

Comment:
Same what's a DestinationIndex? When should I use it? Does it have constraints?

Reply:
added doc